src/main/java/be/acara/events/service/EventServiceImpl.java

```diff
@@ -91,20 +93,33 @@ public Event addEvent(Event event) {
         if (event.getEventDate().isBefore(LocalDateTime.now())) {
             throw new InvalidDateException("Date has to be in the present or future");
         }
-        
         String youtubeUrl = event.getYoutubeId();
-        
-        if (youtubeUrl != null && !youtubeUrl.isEmpty()) {
-            String[] youtubeIdArray = youtubeUrl.split("=");
-            if (youtubeIdArray.length < 2) {
-                throw new InvalidYoutubeUrlException("Invalid youtube URL");
-            }
-            String youtubeId = youtubeIdArray[1];
-            event.setYoutubeId(youtubeId);
-        }
+        event.setYoutubeId(getYoutubeId(youtubeUrl));
         return eventRepository.saveAndFlush(event);
     }
-    
+
+    /**
+     * private method to receive the youtube id from the url
+     *
+     * @param youtubeUrl the youtube url that was given to the event
+     * @return the image id so it can be put in the iframe
+     */
+    private String getYoutubeId(String youtubeUrl) {
+        String youtubeId = "";
+            if (youtubeUrl != null && youtubeUrl.matches("^(http(s)?://)?((w){3}.)?youtu(be|.be)?(\\.com)?/.+")) {
+                Pattern compiledPattern = Pattern.compile("(?<=watch\\?v=|/videos/|embed/|youtu.be/|/v/|/e/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\\\\u200C\\\\u200B2F|youtu.be%2F|%2Fv%2F)[^#&?\\n]*", Pattern.CASE_INSENSITIVE);
+                Matcher matcher = compiledPattern.matcher(youtubeUrl);
+                if (matcher.find()) {
+                    youtubeId = matcher.group();
+                }
+                else {
+                    throw new InvalidYoutubeUrlException("Youtube url is not valid");
+                }
+            }
+        return youtubeId;
```

Comment:
Indentation is off on the `if (youtubeUrl...` block

Remove the else block, instead opt to `return matcher.group();` on line 143 instead. Throw the exception all the way at the end.

I'm not 100% sure but it certainly feels like it but wouldn't this be a method better fit for the Event class?

```
setYoutubeId(String youtubeUrl) {
    // here this method, instead of returning it, setting it, whilst still throwing the exception should it fail
}
```